### PR TITLE
Interleaved add improvements to config

### PR DESCRIPTION
**Pull request by aklife97**
Add improvements to training config for GPT functional tests

### CODE REVIEW

File: launcher_scripts/conf/training/gpt3/175b.yaml [hunk 1]
```diff
@@ -117,13 +117,16 @@ model:
...
+  ub_tp_comm_overlap: True
 
   # miscellaneous
   seed: 1234
   sync_batch_comm: False
   use_cpu_initialization: False # Init weights on the CPU (slow for large models)
   onnx_safe: False # Use work-arounds for known problems with Torch ONNX exporter.
   apex_transformer_log_level: 30 # Python logging level displays logs with severity greater than or equal to this
+  overlap_p2p_comm: True # Overlap p2p communication with computes
+  batch_p2p_comm: False # Batch consecutive inter-peer send/recv operations
```
Comment (ericharper) [inline launcher_scripts/conf/training/gpt3/175b.yaml:129]: can you add the `gc_interval` config as well?

Comment (ericharper) [reply launcher_scripts/conf/training/gpt3/175b.yaml:129]: also this config isn't using transformer engine

Comment (aklife97) [reply launcher_scripts/conf/training/gpt3/175b.yaml:129]: updated all configs

File: launcher_scripts/conf/training/gpt3/175b.yaml [hunk 2]
```diff
@@ -51,7 +51,7 @@ model:
   global_batch_size: 2048
   tensor_model_parallel_size: 8
   pipeline_model_parallel_size: 16
-  virtual_pipeline_model_parallel_size: 6 # interleaved pipeline
+  virtual_pipeline_model_parallel_size: 6 # interleaved pipeline, set to maximum
```
Comment (ericharper) [inline launcher_scripts/conf/training/gpt3/175b.yaml:54]: this needs to be equal to the number of pipeline layers per pipeline rank

Comment (ericharper) [reply launcher_scripts/conf/training/gpt3/175b.yaml:54]: which it is already :D

File: launcher_scripts/conf/training/gpt3/1b_improved.yaml [hunk 3]
```diff
@@ -124,6 +123,20 @@ model:
   fp8_amax_compute_algo: most_recent
   reduce_amax: true
   use_emha: false
+  gc_interval: 100 # Interval of the host memory garbage collection
+
+  ## Transformer Engine
+  transformer_engine: False # leaving this test without TE for fuctional test variety
```
Comment (ericharper) [inline launcher_scripts/conf/training/gpt3/1b_improved.yaml:129]: we can't currently use TE with the improved configs

File: launcher_scripts/conf/training/gpt3/400m_improved.yaml [hunk 4]
```diff
@@ -124,6 +123,20 @@ model:
   fp8_amax_compute_algo: most_recent
   reduce_amax: true
   use_emha: false
+  gc_interval: 100 # Interval of the host memory garbage collection
+
+  ## Transformer Engine
+  transformer_engine: True
```
Comment (ericharper) [inline launcher_scripts/conf/training/gpt3/400m_improved.yaml:129]: needs to be false for improved  configs

File: launcher_scripts/conf/training/gpt3/40b_improved.yaml [hunk 5]
```diff
@@ -124,6 +123,22 @@ model:
   fp8_amax_compute_algo: most_recent
   reduce_amax: true
   use_emha: false
+  overlap_p2p_comm: False # Overlap p2p communication with computes
+  batch_p2p_comm: True # Batch consecutive inter-peer send/recv operations
+  gc_interval: 100 # Interval of the host memory garbage collection
+  
+  ## Transformer Engine
+  transformer_engine: True
```
Comment (ericharper) [inline launcher_scripts/conf/training/gpt3/40b_improved.yaml:131]: false for improved

File: launcher_scripts/conf/training/gpt3/7b_improved.yaml [hunk 6]
```diff
@@ -124,6 +123,20 @@ model:
   fp8_amax_compute_algo: most_recent
   reduce_amax: true
   use_emha: false
+  gc_interval: 100 # Interval of the host memory garbage collection
+
+  ## Transformer Engine
+  transformer_engine: True
```
Comment (ericharper) [inline launcher_scripts/conf/training/gpt3/7b_improved.yaml:129]: false for improved

**Review by ericharper — CHANGES_REQUESTED** [review]
Few comments

**Comment by aklife97** [comment]
PR contains non-optimal settings for some configs for diversity of testing, all these params are marked with comments in front of them
The optimal settings are present in #61  for perf tests

File: launcher_scripts/conf/training/gpt3/1b_improved.yaml (hunk 3 above)
Comment (aklife97) [reply launcher_scripts/conf/training/gpt3/1b_improved.yaml]: done

File: launcher_scripts/conf/training/gpt3/40b_improved.yaml (hunk 5 above)
Comment (aklife97) [reply launcher_scripts/conf/training/gpt3/40b_improved.yaml]: done

File: launcher_scripts/conf/training/gpt3/7b_improved.yaml (hunk 6 above)
Comment (aklife97) [reply launcher_scripts/conf/training/gpt3/7b_improved.yaml]: done

File: launcher_scripts/conf/training/gpt3/400m_improved.yaml (hunk 4 above)
Comment (aklife97) [reply launcher_scripts/conf/training/gpt3/400m_improved.yaml]: done

**Review by ericharper — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**